autochdir.kak

```diff
@@ -0,0 +1,38 @@
+##
+## autochdir.kak by wheatdog
+## Automatically change server's working directory according to the current focus buffer
+## To turn on this option, add `set global autochdir true` in your kakrc
+##
+
+decl bool autochdir
+decl str working_folder
+
+def -hidden \
+    autochdir-wrapper %{ %sh{
+        if [ "${kak_opt_autochdir}" == "true" ] && [ ! -z "${kak_opt_working_folder}" ] && [ -d "${kak_opt_working_folder}" ]; then
+            echo "cd ${kak_opt_working_folder}" 
+        fi
+} }
+
+def -shell-candidates %{ls -d */} -params ..1 -docstring %{change-directory! [<directory>] : like change-directory, but also update the buffer's working directory} \
```

Comment:
I'd feel more comfortable with a `find -type d` rather than the `ls` and `-params ..1` can be replaced with `-params 1`.